### PR TITLE
Qute API cleanup - get rid of the "io.quarkus.qute.api" package

### PR DESCRIPTION
**Pull request by mkouba**
- deprecate io.quarkus.qute.api.ResourcePath and replace it with
io.quarkus.qute.Location
- deprecate io.quarkus.qute.api.CheckedTemplate and replace it with
io.quarkus.qute.CheckedTemplate

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Qualifies an injected template. The {@link #value()} is used to locate the template; it represents the path relative from
+ * the base path.
+ */
+@Retention(RUNTIME)
+@Target({ FIELD, PARAMETER, METHOD })
+public @interface Location {
```
Comment (FroMage) [inline independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: Are you sure about the name? Sounds a bit too generic, I can expect other libs will use the same name, no? What about `@TemplatePath`?

Comment (mkouba) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: Well, I'm not aware of any other `Location` annotation. The problem with `@TemplatePath` is that an injection point would look like `@TemplatePath("/foo/bar") Template bar`, i.e. too many templates. It also corresponds with the [TemplateLocator](https://github.com/quarkusio/quarkus/blob/master/independent-projects/qute/core/src/main/java/io/quarkus/qute/TemplateLocator.java) API.

Comment (mkouba) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: That said I'm not sure at all ;-)

Comment (FroMage) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: Me neither. Probably we should ask for feedback about this.

Comment (gastaldi) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: `@TemplateFile`? :)

**Comment by mkouba** [comment]
Note that `@io.quarkus.qute.Location` is only used if you need to inject a template with a name that can't be used as a Java field or a template from a nested dir, i.e. `src/main/resources/templates/foo/bar/baz.html`. It's not needed for a template with a "common" name from the root dir and of course, it's not needed for templates used via `@CheckedTemplate`.

As for the name - I don't think it needs to contain the `Template` str because we're injecting a `io.quarkus.qute.Template` anyway. I suggested `@Location` because (1) it's neutral (i.e. it can reference templates outside `src/main/resources/templates` in the future), (2) it corresponds to the `io.quarkus.qute.TemplateLocator` API and (3) it's short and concise.

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java (hunk 1 above)
Comment (geoand) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: `@Resource`?

Comment (mkouba) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: > `@Resource`?

`javax.annotation.Resource` and `jakarta.annotation.Resource`...

Comment (geoand) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: :(

Comment (gastaldi) [reply independent-projects/qute/core/src/main/java/io/quarkus/qute/Location.java:17]: `@TemplateLocation`?

**Comment by maxandersen** [comment]
btw. how come we don't want the qute.api package ? refreshing to actually be able to identify what is API and what is not API :)

**Comment by mkouba** [comment]
> btw. how come we don't want the qute.api package ? refreshing to actually be able to identify what is API and what is not API :)

The general consensus in quarkus is that we don't use the `*.api` package. Normally, the API should be all `public` stuff from the `io.quarkus.qute` package but since Qute is an "independent" project this package is already taken.

**Pull request merged**